pkg/controller/configmap_watcher.go

```diff
@@ -14,13 +16,28 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
+// this structure mirrors Kube-Linter configuration structure
+// it is used as a bridge to unmarshall ConfigMap data
+// doc: https://pkg.go.dev/golang.stackrox.io/kube-linter/pkg/config#Config
+type KubeLinterChecks struct {
+	Checks struct {
+		AddAllBuiltIn        bool     `yaml:"addAllBuiltIn,omitempty"`
+		DoNotAutoAddDefaults bool     `yaml:"doNotAutoAddDefaults,omitempty"`
+		Exclude              []string `yaml:"exclude,omitempty"`
+		Include              []string `yaml:"include,omitempty"`
+		IgnorePaths          []string `yaml:"ignorePaths,omitempty"`
+	} `yaml:"checks"`
+}
+
 type ConfigMapWatcher struct {
 	clientset kubernetes.Interface
-	ch        chan struct{} // TODO - TBD configmap struct
+	checks    KubeLinterChecks
+	ch        chan config.Config
 }
 
-var configMapName = "deployment-validation-operator-config"
+var configMapName = "deployment-validation-operator-custom-config"
```

Comment:
Maintaining compatibility on previous ConfigMap name. It's already mounted in a volume and it's redundant to get it from the API.

Reply:
I think I don't understand the reason behind this rename.

Reply:
Currently default ConfigMap is part of the Deployment and we cannot change it. So, this adds a custom ConfigMap which data will be prioritized over the default one (immutable). And I added `custom` to the ConfigMap name to not overlap the basic, default one.